chart/k8gb/templates/_helpers.tpl

```diff
@@ -156,3 +162,24 @@ k8gb-{{ .Values.k8gb.dnsZone }}-{{ .Values.k8gb.clusterGeoTag }}
 {{- define "k8gb.metrics_port" -}}
 {{ print (split ":" .Values.k8gb.metricsAddress)._1 }}
 {{- end -}}
+
+{{- define "external-dns.azure-credentials" -}}
+{{- if .Values.azuredns.enabled -}}
+{
+  "tenantId": "{{ .Values.azuredns.tenantId }}",
+  "subscriptionId": "{{ .Values.azuredns.subscriptionId }}",
+  "resourceGroup": "{{ .Values.azuredns.resourceGroup }}",
+  {{- if .Values.azuredns.aadClientId -}}
+  "aadClientId": "{{ .Values.azuredns.aadClientId }}",
+  {{- end -}}
+  {{- if .Values.azuredns.aadClientSecret -}}
+  "aadClientSecret": "{{ .Values.azuredns.aadClientSecret }}",
```

Comment:
Does it mean we will have the client secret as plain text in the `values.yaml`? We probably should keep it in the `Secret`